pd/src/info.rs

```diff
@@ -56,9 +58,48 @@ impl Info {
         &self,
         query: abci::request::Query,
     ) -> Result<abci::response::Query, anyhow::Error> {
-        tracing::warn!(?query, "unhandled query");
+        tracing::info!(?query);
+
+        match query.path.as_str() {
+            "/jmt/key" => {
+                let state = self.state_tonic().await?;
+                let (value, proof) = state
+                    .read()
+                    .await
+                    .get_with_proof(query.data.to_vec())
+                    .await
+                    .map_err(|e| tonic::Status::internal(e.to_string()))?;
+
+                let commitment_proof = ics23::CommitmentProof {
+                    proof: Some(ics23::commitment_proof::Proof::Exist(proof)),
+                };
+
+                let kvr = KeyValueResponse {
+                    value,
+                    proof: Some(commitment_proof),
+                };
+
+                let height = state.get_block_height().await?;
+
+                Ok(abci::response::Query {
+                    code: 0,
+                    key: query.data,
+                    log: "".to_string(),
+                    value: kvr.encode_to_vec().into(),
+                    // NOTE: the ABCI query proof is not the same as ICS-23 proofs.
+                    proof: None,
```

Comment:
this is a bit ugly. the ABCI query interface includes a `proof` field, but it's a tendermint::merkle:Proof, which is *not* the same as an ics23 proof (and not as general as an ics23 proof). SDK chains actually have to convert this to an ics23 proof to work. Instead, we don't use this field, and instead directly include the ics23 proof in the `value` field of the query (encoded into the KeyValueResponse).

Reply:
Hmm, that is ugly. Maybe we can get them to fix this upstream? It'd be a breaking change, but getting it into the ecosystem at some poitn seem better...